src/common/users.ts

```diff
@@ -121,16 +169,37 @@ export const getUserReadingRank = async (
       'lastReadTime',
     )
     .from(View, 'view')
-    .where('"userId" = :id', { id: userId })
-    .getRawOne<ReadingRankQueryResult>();
-  const rankThisWeek = rankFromProgress(res.thisWeek);
-  const rankLastWeek = rankFromProgress(res.lastWeek);
+    .where('"userId" = :id', { id: userId });
+
+  const getReadingTags = () => {
+    if (!includeTags) {
+      return Promise.resolve(null);
+    }
+
+    const now = Date.now();
+    const start = new Date(startOfWeek(now).getTime()).toISOString();
+    const end = new Date(endOfWeek(now).getTime()).toISOString();
+
+    return getUserReadingDays(con, {
+      userId,
+      timezone,
+      dateRange: { start, end },
+    });
+  };
+
+  const [{ thisWeek, lastWeek, lastReadTime }, tags] = await Promise.all([
+    req.getRawOne<ReadingRankQueryResult>(),
+    getReadingTags(),
+  ]);
+  const rankThisWeek = rankFromProgress(thisWeek);
+  const rankLastWeek = rankFromProgress(lastWeek);
   return {
-    lastReadTime: res.lastReadTime,
+    lastReadTime: lastReadTime,
     currentRank: rankThisWeek > rankLastWeek ? rankThisWeek : rankLastWeek,
-    progressThisWeek: res.thisWeek,
+    progressThisWeek: thisWeek,
     rankLastWeek,
     rankThisWeek,
-    readToday: isSameDay(res.lastReadTime, new Date()),
+    readToday: isSameDay(lastReadTime, new Date()),
```

Comment:
You can take the `now` definition outside of the function scope and utilize here as well :)